environment.yml

```diff
@@ -37,7 +37,7 @@ dependencies:
   - packaging>=20
   - pydata-sphinx-theme~=0.15.0
   - pyyaml
-  - sphinx>=3.0.0,!=6.1.2
+  - sphinx>=7.4.0,!=6.1.2
```

Comment:
Not sure we want to be that aggressive. On a technical side, it's ok. The only concern is that we break every dev envionment out there. When this is in, if somebody does a rebase onto master, their sphinx-build will fail because their dev env has an older version of sphinx and cannot handle `:class:` in `.. toctree:`. I'd slightly prefer to still go with the `.. rst-class` variant for now, and change some time in the future.

Sorry if my comment has made you believe I advocate for switching right now. But please decide yourself.